src/v/kafka/client/cluster.cc

```diff
@@ -236,35 +254,24 @@ ss::future<> cluster::dispatch_metadata_request(
           });
     }
 
-    try {
-        auto broker = _brokers.any();
-        // Initialize the broker if it is not connected, this is required to
-        // make sure the broker has up to date API version information.
-
-        auto request_version = co_await get_metadata_request_version(
-          broker, _as);
-        auto metadata_version
-          = requested_version.has_value()
-              ? std::min(requested_version.value(), request_version)
-              : request_version;
-        // TODO: support topic subscription
-        auto reply = co_await broker->dispatch(
-          metadata_request{.data{
-            .topics = std::move(topics_to_request),
-            .allow_auto_topic_creation = false,
-            .include_topic_authorized_operations = true}},
-          metadata_version);
-        vassert(
-          std::holds_alternative<kafka::metadata_response>(reply),
-          "Metadata response is required to be returned as a result of "
-          "metadata request");
-
-        co_await apply_metadata(to_metadata_update(
-          std::get<kafka::metadata_response>(std::move(reply))));
-    } catch (const broker_error& e) {
-        vlog(
-          _logger.warn, "Failed to dispatch metadata request - {}", e.what());
-    }
+    auto request_version = co_await get_metadata_request_version(broker, _as);
+    auto metadata_version = requested_version.has_value()
+                              ? std::min(
+                                  requested_version.value(), request_version)
+                              : request_version;
```

Comment:
I wonder if it's worth passing the `requested_version` into `get_metadata_request_version` if this is going to be a common pattern for other requests.

nitpick:
```suggestion
    auto metadata_version = std::min(
      requested_version.value_or(request_version), request_version);
```